_posts/jamesma560/2016-04-06-MOAR-Dictionary-Exercises.md

```diff
@@ -0,0 +1,26 @@
+---
+author: jamesma560
+layout: post
+title: "James Moore MOAR Dictionary Exercises"
+---
+
+Exercise 1:
+
+<iframe src="https://trinket.io/embed/python3/f210092c0d" width="100%" height="356" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+Exercise 2-5:
+
+<iframe src="https://trinket.io/embed/python3/c95634db14" width="100%" height="356" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+These exercises were much easier having read the regex chapter. At first, I was hesitant
+to implement regex in my code because I wasn't confident enought that I could come up
+with viable regex expressions in an effiicient enough manner. However, I found a nifty website (https://regex101.com/) that 
+allowed me to test differet regexes outside the scope of the rest of my code. Therefore I could just copy and paste a line
+from mbox-short.txt into regex101 to test out regexes to see if they worked. 
+
+This exercise also hammered home how to access the key of a dictionary as opposed to its value. For whatever reason
+some of the stumbling blocks I encountered involved accessing the value when I meant to access the key and vice versa. With this 
+practice using for loops to navigage and build dictionaries, it really helped me understand dictionaries a little bit more.
+
+But despite this added knowledge I was unable to figure out exercise 5. Really it came down to the fact that I couldn't build
+a regex to capture just the domain name. I'll be interested to see how you figured this out. 
```

Comment:
The parentheses in a regex extract what you need.  In an email address the domain is every non-whitespace character between `@` and whitespace.  So `^From\s\S*@(\S*)` pulls it out (and there are others which will too.)